File: DemoMethods/Advanced/FullTextSearch.cs
using System;
using System.Linq;
using System.Web.Http;
using DemoMethods.Entities;
using DemoMethods.Helpers;
using DemoMethods.Indexes;
using Raven.Client;

namespace DemoMethods.Advanced
{
    public partial class AdvancedController : DemoApiController
    {
        [HttpGet]
        [Demo("Full Text Search", DemoOutputType.Flatten, demoOrder: 200)]
        public object FullTextSearch(string searchTerm = "Jazz")
        {
            using (var session = DocumentStoreHolder.MediaStore.OpenSession())
            {
                RavenQueryStatistics stats;
                var results = session.Query<LastFmAnalyzed.Result, LastFmAnalyzed>()
                    .Take(20)
                    .Search(x => x.Query, searchTerm)
                    .TransformWith<TransformerLastFm, LastFm>()
                    .Statistics(out stats)
                    .ToList();

                ServerTime = TimeSpan.FromMilliseconds(stats.DurationMilliseconds);

                return results;
            }
        }
    }
}

File: DemoMethods/PythonDemo/demo_methods/Advanced/FullTextSearch.py
from demo.holder import DocumentStoreHolder
from datetime import timedelta
from demo.entities import LastFm
from demo.indexes import LastFmAnalyzed


class FullTextSearch(object):
    @staticmethod
    def full_text_search(searchTerm="Jazz"):
        with DocumentStoreHolder.get_store().open_session() as session:
            query, stats = list(session.query(object_type=LastFm, with_statistics=True,
                                              index_name=LastFmAnalyzed.__name__).
                                search("Query", search_terms=searchTerm))

            server_time = timedelta(milliseconds=stats["DurationMilliseconds"])

            return query
